test/remittance.js

```diff
@@ -4,7 +4,267 @@ const Remittance = artifacts.require("./Remittance.sol");
+        );
+        checkEventNotEmitted();
+    });
+
+    it("Withdrawing funds reverts when amount is zero", async () => {
+
+        await remittance.deposit({from: funder, value: 5});
+
+        await truffleAssert.reverts(
+            remittance.withdraw(0, {from: funder}),
+            "The value must be greater than 0"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("Withdrawing funds reverts when the withdrawer has insufficient funds", async () => {
+
+        await truffleAssert.reverts(
+            remittance.withdraw(1, {from: broker}),
+            "There are insufficient funds"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("Remittance can only be paused by the owner", async () => {
+
+        await truffleAssert.reverts(
+            remittance.pause({from: broker}),
+            "Ownable: caller is not the owner"
+        );
+
+        await truffleAssert.reverts(
+            remittance.unpause({from: broker}),
+            "Ownable: caller is not the owner"
+        );
+    });
+
+    it("Deposit is pausable and unpausable", async () => {
+
+        await remittance.pause({from: funder});
+
+        await truffleAssert.reverts(
+            remittance.deposit({from: funder, value: 5}),
+            "Pausable: paused"
+        );
+        checkEventNotEmitted();
+
+        await remittance.unpause({from: funder});
+        const txObj = await remittance.deposit({from: funder, value: 5});
+
+        truffleAssert.eventEmitted(txObj, 'Deposit');
+    });
+
+    it("Create is pausable and unpausable", async () => {
+
+        await remittance.deposit({from: funder, value: 5}),
+        await remittance.pause({from: funder});
+
+        await truffleAssert.reverts(
+            remittance.create(hash, broker, 2, {from: funder}),
+            "Pausable: paused"
+        );
+        checkEventNotEmitted();
+
+        await remittance.unpause({from: funder});
+        const txObj = await remittance.create(hash, broker, 2, {from: funder});
+
+        truffleAssert.eventEmitted(txObj, 'RemittanceCreated');
+    });
+
+    it("Release is pausable and unpausable", async () => {
+
+        await remittance.deposit({from: funder, value: 5}),
+        await remittance.create(hash, broker, 5, {from: funder});
+        await remittance.pause({from: funder});
+
+        await truffleAssert.reverts(
+            remittance.release(concatPassword, {from: broker}),
+            "Pausable: paused"
+        );
+        checkEventNotEmitted();
+
+    });
+
+    it("Withdraw is pausable and unpausable", async () => {
+
+        await remittance.deposit({from: funder, value: 5});
+        await remittance.pause({from: funder});
+
+        await truffleAssert.reverts(
+            remittance.withdraw(1, {from: funder}),
+            "Pausable: paused"
+        );
+        checkEventNotEmitted();
+    });
 
 });
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
That's some serious space for future tests 😂

Reply:
Oh. Yes!